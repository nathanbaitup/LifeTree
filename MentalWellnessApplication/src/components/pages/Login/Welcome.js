import React, { useState } from 'react';
import { Text, View, ImageBackground, TextInput, Image, TouchableOpacity } from 'react-native';
import Loading from '../../utils/Loading';

// Imports the documents styling.
import { loginStyles } from './Styles';

// Imports authentication and firestore from firebase
import auth from '@react-native-firebase/auth';
import firestore from '@react-native-firebase/firestore';


export default function Welcome({ navigation }) {
    // REFERENCE ACCESSED 13/12/2021 https://www.freecodecamp.org/news/react-native-firebase-tutorial/
    // Used to integrate firebase authentication into the application to create and sign into an account.
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [loading, setLoading] = useState(false);

    const onLoginPress = () => {
        setLoading(true);
        // Calls the signInWithEmailAndPassword API from Auth to take the email and password entered and check if the user exists.
        // If successful, the user data is stored to the variable 'user' to be parsed through the application and the user
        // is signed in and taken to the home page.
        auth()
            .signInWithEmailAndPassword(email, password)
            .then((response) => {
                const uid = response.user.uid;
                const usersRef = firestore().collection('users');
                usersRef
                    .doc(uid)
                    .get()
                    .then(firestoreDocument => {
                        if (!firestoreDocument.exists) {
                            alert('User does not exist. Please check your credentials.');
                            return;
                        }
                        setLoading(false);
                        navigation.navigate('HomeScreen');
                    })
                    .catch(error => {
                        setLoading(false);
                        alert(error);
                    });
            })
            .catch(error => {
                setLoading(false);
                alert(error);
            });
    };
    // END REFERENCE

    if (loading) {
        return (
            <Loading loading={loading} />
        );
    } else {
        return (
            <ImageBackground source={require('../../../resources/img/background.png')} style={{ width: '100%', height: '100%', opacity: 50 }} >
                <View style={loginStyles.header}>
                    <Text style={loginStyles.mainTitle}>Welcome to LifeTree</Text>
                    <Text>Log into your account below:</Text>
                </View>
                <View style={loginStyles.contentContainer}>
                    <Image style={loginStyles.logo} source={require('../../../resources/img/faces/happy.png')} />

                    <TextInput
                        style={loginStyles.textInput}
                        placeholder='enter your email address...'
                        placeholderTextColor='#aaaaaa'
                        onChangeText={(textEmail) => setEmail(textEmail)}
                        value={email}
                        autoCapitalize='none'
                    />

                    <TextInput
                        style={loginStyles.textInput}
                        secureTextEntry
                        placeholder='enter your password...'
                        placeholderTextColor='#aaaaaa'
                        onChangeText={(textPass) => setPassword(textPass)}
                        value={password}
                        autoCapitalize='none'
                    />

                    <TouchableOpacity
                        style={loginStyles.loginBTN}
                        onPress={() => onLoginPress()}>
                        <Text style={loginStyles.loginText}> Log In </Text>
                    </TouchableOpacity>

                    <View style={loginStyles.footer}>
                        <Text style={loginStyles.createAnAccountText} >Don&apos;t have an account yet?<Text onPress={() => navigation.navigate('CreateAccount')} style={loginStyles.createAccountLink}> Sign up</Text> </Text>
                    </View>
                </View>
            </ImageBackground>
        );
    }
}
