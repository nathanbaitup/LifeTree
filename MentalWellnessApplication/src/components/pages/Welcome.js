import React  from 'react';
import { Text, View, Button, ImageBackground, StyleSheet } from 'react-native';

export default function Welcome({navigation}) {
    return (
        <ImageBackground source={require('../resources/img/background.png')} style={{ width: '100%', height: '100%', opacity: 50 }} >
            <View style={styles.header}>
                <Text style={styles.mainTitle}>Welcome to LifeTree </Text>
                <Text>What would you like to do?</Text>
            </View>
            <View style={styles.contentContainer}>
                <Button title="Login" />
                <Button title="Sign Up" />
                <Button title="Continue as Guest" onPress={() => navigation.navigate('Home')} />
            </View>
        </ImageBackground >
    );
}


const styles = StyleSheet.create({

    header: {
        justifyContent: 'center',
        alignItems: 'center',
        paddingTop: 50,
    },
    mainTitle: {
        fontSize: 25,
        fontWeight: 'bold',
        textAlign: 'center',
    },
    contentContainer: {
        margin: 20,
        padding: 10,
        backgroundColor: '#FFFFFF',
        borderRadius: 25,
    },
});
