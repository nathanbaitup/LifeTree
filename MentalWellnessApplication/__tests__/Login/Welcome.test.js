import 'react-native';
import React from 'react';
import Welcome from '../../src/components/pages/Login/Welcome';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Checks to see if the Welcome page renders', () => {
  const tree = renderer.create(<Welcome />).toJSON();
  expect(tree).toMatchSnapshot();
});