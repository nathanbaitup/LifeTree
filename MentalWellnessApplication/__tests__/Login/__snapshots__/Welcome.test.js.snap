// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks to see if the Welcome page renders 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "opacity": 50,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/resources/img/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingTop": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Welcome to LifeTree
    </Text>
    <Text>
      Log into your account below:
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 25,
        "justifyContent": "center",
        "margin": 20,
        "padding": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/resources/img/faces/happy.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 100,
          "marginBottom": 10,
          "marginTop": 20,
          "width": 134,
        }
      }
    />
    <TextInput
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="enter your email address..."
      placeholderTextColor="#aaaaaa"
      style={
        Object {
          "backgroundColor": "#f0efed",
          "borderRadius": 5,
          "height": 50,
          "marginBottom": 10,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 10,
          "overflow": "hidden",
          "paddingLeft": 10,
        }
      }
      value=""
    />
    <TextInput
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="enter your password..."
      placeholderTextColor="#aaaaaa"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#f0efed",
          "borderRadius": 5,
          "height": 50,
          "marginBottom": 10,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 10,
          "overflow": "hidden",
          "paddingLeft": 10,
        }
      }
      value=""
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#00e676",
          "borderRadius": 15,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "padding": 15,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
         Log In 
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        Don't have an account yet?
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#6db8bf",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
           Sign up
        </Text>
         
      </Text>
    </View>
  </View>
</View>
`;
